### PR TITLE
Fix travis.yml and test_mpfr_trig for osx.

### PR DESCRIPTION
**Pull request by vinklein**
rewrite test_mpfr_trig:  mpmath is platform dependent and using it cause error with osx like in this build "https://travis-ci.org/vinklein/gmpy/jobs/297897566". These errors appear in issue #163 too.
Doctest on osx works with this version (https://travis-ci.org/vinklein/gmpy/jobs/298538269).

travis.yml: Rework the file to have osx tests running correctly (cython and make docs). 
Linked to issue #157.
still to fix : make "make latexpdf" command work on osx

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,30 +1,82 @@
...
+      echo ""
+      echo "Installing latex"
+      port install texlive texlive-latex-extra
+      python setup.py install --shared=$CONDA_PREFIX;
+    else # Linux
+      export PATH="/usr/lib/ccache:$PATH"
+      ccache -M 256M && ccache -s
+      sudo apt-get install libgmp-dev libmpfr-dev libmpfi-dev libmpc-dev cython texlive texlive-latex-extra latexmk
+      # note: sphinx in ubuntu is version 1.2.2 that does not support
+      # sphinx.ext.imgmath. We just install the last version.
```
Comment (videlec) [inline .travis.yml]: This comment about sphinx is now at the wrong place

File: .travis.yml [hunk 2]
```diff
@@ -1,30 +1,82 @@
...
+  - |
+    if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+      echo ""
+      echo "Installing a fresh version of Miniconda."
+      MINICONDA_URL="https://repo.continuum.io/miniconda"
+      MINICONDA_FILE="Miniconda3-latest-MacOSX-x86_64.sh"
+      curl -L -O "${MINICONDA_URL}/${MINICONDA_FILE}"
+      bash $MINICONDA_FILE -b
+      # Configure conda.
+      echo ""
```
Comment (videlec) [inline .travis.yml]: idem

File: .travis.yml [hunk 3]
```diff
@@ -1,30 +1,82 @@
...
+      echo "Configuring conda."
+      source $HOME/miniconda3/bin/activate root
+      conda config --add channels conda-forge
+      conda config --set show_channel_urls true
+      conda config --set always_yes yes --set changeps1 no
+      echo ""
```
Comment (videlec) [inline .travis.yml]: idem

File: .travis.yml [hunk 4]
```diff
@@ -1,30 +1,82 @@
...
+      echo "Creating conda environment"
+      conda create -n test python=${PYTHON_VERSION} gmp mpfr mpc
+      source activate test
+      echo ""
```
Comment (videlec) [inline .travis.yml]: idem

File: .travis.yml [hunk 5]
```diff
@@ -1,30 +1,82 @@
...
+
+    - os: osx
+      env: PYTHON_VERSION=3.6
+      osx_image: xcode6.4
+
 install:
-  - pip install --verbose .
+  - |
+    if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+      echo ""
```
Comment (videlec) [inline .travis.yml]: What is the point of `echo ""`?

File: .travis.yml [hunk 6]
```diff
@@ -1,30 +1,82 @@
...
+      export PATH="/usr/lib/ccache:$PATH"
+      ccache -M 256M && ccache -s
+      sudo apt-get install libgmp-dev libmpfr-dev libmpfi-dev libmpc-dev cython texlive texlive-latex-extra latexmk
+      # note: sphinx in ubuntu is version 1.2.2 that does not support
+      # sphinx.ext.imgmath. We just install the last version.
+      pip install --verbose .
+    fi
+
 script:
+  - pip install sphinx
```
Comment (videlec) [inline .travis.yml]: this should not be here. `pip install sphinx` is part of the installation process.

**Comment by videlec** [comment]
Did you open an issue for latexpdf on osx or do you mean you will solve it in this PR?

File: .travis.yml (hunk 5 above)
Comment (vinklein) [reply .travis.yml]: skipping a line in the trace. I think the point is readability.

File: .travis.yml (hunk 6 above)
Comment (vinklein) [reply .travis.yml]: right

**Comment by vinklein** [comment]
I will dig in the problem, and open an issue if i am stuck. Either way i don't think this should block this PR (as osx test take so long to start on travis)

File: .travis.yml (hunk 5 above)
Comment (videlec) [reply .travis.yml]: Readability of what? It does not help reading the script. And on travis you have access to the output of each command separatly.

File: .travis.yml [hunk 7]
```diff
@@ -1,30 +1,82 @@
...
+      echo "Installing latex"
+      port install texlive texlive-latex-extra
+      python setup.py install --shared=$CONDA_PREFIX;
+    else # Linux
+      export PATH="/usr/lib/ccache:$PATH"
+      ccache -M 256M && ccache -s
+      sudo apt-get install libgmp-dev libmpfr-dev libmpfi-dev libmpc-dev cython texlive texlive-latex-extra latexmk
+      pip install --verbose .
+    fi
+    # note: sphinx in ubuntu is version 1.2.2 that does not support
```
Comment (videlec) [inline .travis.yml]: This remark is not appropriate anymore since it applies to both Ubuntu and OSX.

File: .travis.yml (hunk 5 above)
Comment (vinklein) [reply .travis.yml]: Readability of the log on travis console.
With the '|' char allowing multiple instructions, the whole block (if to fi in this instance) is considered as a single command by the logs.

**Comment by casevh** [comment]
Thanks!

**Pull request merged**